projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -139,6 +192,10 @@ export default function MediaSection() {
 
 	const renderPreview = useCallback(
 		open => {
+			if ( ! Object.keys( mediaData ).length || ! Object.keys( metaData ).length ) {
+				return null;
+			}
+
```

Comment:
We might be able to simplify this. The `mediaData` checks are handled below. Depending on how things are handled, we could leave metaData unchecked, or if it ends up being undefined rather than an empty object it could be a `! metaData` check.